app/controllers/errors_controller.rb

```diff
@@ -0,0 +1,30 @@
+class ErrorsController < ApplicationController
+  layout 'errors'
+
+  SUPPORTED_STATUS_CODES = [400, 404, 406, 422, 500].freeze
+
+  def show
+    @error = request.env['action_dispatch.exception']
+    @status_code = determine_status_code
+
+    respond_to do |format|
+      format.html { render "errors/#{@status_code}", status: @status_code }
+      format.any { head @status_code }
+    end
+  end
+
+  private
+
+  def determine_status_code
+    status_from_params = params[:status_code]&.to_i
```

Comment:
What's the case where this will be invalid/nil? In routes.rb, we're matching each of the status codes explicitly so every other code will be a 404.